app/models/name/merge.rb

```diff
@@ -130,4 +134,40 @@ def merge(old_name)
     # Finally destroy the name.
     old_name.destroy
   end
+
+  ##############################################################################
+
+  private
+
+  def approved_synonym_of_proposed_name?
+    !deprecated && (other_synonym_ids & Naming.all.pluck(:name_id)).present?
+  end
+
+  def correctly_spelled_ancestor_of_proposed_name?
+    return false if correct_spelling.present?
+    return above_genus_is_ancestor? if classified_above_genus?
+    return genus_or_species_is_ancestor? if [:Genus, :Species].include?(rank)
+    return group_is_ancestor? if rank == :Group
+
+    false
+  end
+
+  def above_genus_is_ancestor?
+    Name.joins(:namings).where(
+      "classification LIKE ?", "%#{rank}: #{classification_name}%"
+    ).any?
+  end
+
+  def genus_or_species_is_ancestor?
+    Name.joins(:namings).where("text_name LIKE ?", "#{text_name}%").
```

Comment:
Careful here, too.  You really need a trailing space (" ") to ensure "Testgenuswithotherstuff" doesn't match "Testgenus" accidentally.  I recommend either of these

    ...where("text_name LIKE '#{text_name} %'")
    ...where("text_name LIKE '#{text_name} %' OR text_name = '#{text_name}'")

depending on whether the equals case is already handled by Name#mergeable?.

Reply:
Goog catch. Thank you!

Reply:
The first suggestion (`where("text_name LIKE '#{text_name} %'")` is the right one.
- This is independent of `mergeable?` 
- To decide if another Proposed Name "depends" on this Name, we must ignore Namings of this name.